src/main/java/org/opensrp/service/ProductCatalogueService.java

```diff
@@ -33,11 +33,13 @@ public ProductCatalogue getProductCatalogue(Long uniqueId) {
 
 	public void add(ProductCatalogue productCatalogue) {
 		validateFields(productCatalogue);
+		productCatalogue.setServerVersion(productCatalogueRepository.getNextServerVersion());
 		productCatalogueRepository.add(productCatalogue);
 	}
 
 	public void update(ProductCatalogue productCatalogue) {
 		validateFields(productCatalogue);
+		productCatalogue.setServerVersion(productCatalogueRepository.getNextServerVersion());
```

Comment:
You need to define Sequence in productCatalogueRepository for this to work

Reply:
Done